### ISSUE TITLE
Embed's thumbnail_url model field is too short

### ISSUE THREAD

**Issue by GTpyro** [wagtail/wagtail#6467]
### Issue Summary

The thumbnail_url model field in embeds/models.py is too short. Currently, the character limis is set to 255 however some providers return a longer url (e.g. Instagram). This does not affect sqlite backends as sqlite does not enforce field length limits. When this error occurs, it results in the following stack trace:
```
[16/Oct/2020 02:59:41] "GET /admin/embeds/chooser/ HTTP/1.1" 200 1556
Internal Server Error: /admin/embeds/chooser/upload/
Traceback (most recent call last):
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/models/query.py", line 573, in get_or_create
    return self.get(**kwargs), False
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/models/query.py", line 429, in get
    raise self.model.DoesNotExist(
wagtail.embeds.models.Embed.DoesNotExist: Embed matching query does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(255)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 179, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/views/decorators/cache.py", line 44, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/wagtail/admin/urls/__init__.py", line 110, in wrapper
    return view_func(request, *args, **kwargs)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/wagtail/admin/auth.py", line 188, in decorated_view
    return view_func(request, *args, **kwargs)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/wagtail/embeds/views/chooser.py", line 29, in chooser_upload
    embed_html = embed_to_editor_html(form.cleaned_data['url'])
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/wagtail/embeds/format.py", line 21, in embed_to_editor_html
    embed = embeds.get_embed(url)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/wagtail/embeds/embeds.py", line 42, in get_embed
    embed, created = Embed.objects.get_or_create(
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/models/query.py", line 576, in get_or_create
    return self._create_object_from_params(kwargs, params)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/models/query.py", line 610, in _create_object_from_params
    obj = self.create(**params)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/models/query.py", line 447, in create
    obj.save(force_insert=True, using=self.db)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/models/base.py", line 753, in save
    self.save_base(using=using, force_insert=force_insert,
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/models/base.py", line 790, in save_base
    updated = self._save_table(
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/models/base.py", line 895, in _save_table
    results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/models/base.py", line 933, in _do_insert
    return manager._insert(
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/models/query.py", line 1254, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/models/sql/compiler.py", line 1397, in execute_sql
    cursor.execute(sql, params)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 98, in execute
    return super().execute(sql, params)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/someuser/code/wagtail_scratch/bugreport/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.DataError: value too long for type character varying(255)

[16/Oct/2020 02:59:53] "POST /admin/embeds/chooser/upload/ HTTP/1.1" 500 267032
```

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. Edit models.py per the [getting started tutorial](https://docs.wagtail.io/en/v2.10.2/getting_started/tutorial.html)
2. Edit settings/base.py to change the database from sqlite to postgres (likely effects other databases too but I only tested with postgres)
3. Create the database entries with `python manage.py makemigrations` & `python manage.py migrate`
4. Create admin account via `python manage.py createsuperuser`
5. Run the server via `python manage.py runserver`
6. Navigate to localhost:8000/admin/ & login to the wagtail admin console
7. Navigate to the default created page (typically localhost:8000/admin/pages/3/) and click on Add Child Page button
8. In the body of the page, attempt to embed a url by clicking the embed button. For example this one `https://www.instagram.com/p/CF0Wt_oDgeG/?utm_source=ig_web_button_share_sheet`

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?
* Wagtail should be able to handle longer URLs in embeds as the oEmbed standard does not specify a maximum url length. This likely impacts the url and thumbnail_url fields.

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

* Python version: 3.8.3
* Django version: 3.1.2 & 3.0.8
* Wagtail version: 2.10.10
* Browser version: chrome 85.0.4183.102

**Comment by th3hamm0r** [comment]
We've also run into this issue with long `thumbnail_urls` from Instagram. Until this issue is fixed we're using this signal handler to clear those urls:

    from django.db.models.signals import pre_save
    from django.dispatch import receiver
    from wagtail.embeds.models import Embed


    @receiver(pre_save, sender=Embed)
    def fix_embed_thumbnail_url(sender, instance: Embed, **kwargs):
        if instance.thumbnail_url and len(instance.thumbnail_url) > 255:
            instance.thumbnail_url = None

**Comment by originell** [comment]
We've also run into this issue. We usually want to keep the thumbnail URL around. Thankfully, we are running Postgresql, so having a charfield with >255 characters is possible. So we resorted to monkey patch:

```python
# some __init__.py. e.g. `pages/models/__init__.py`
from wagtail.embeds.models import Embed

# Instagram returns URLs longer than 255 chars suddenly. 
# Sadly, Wagtail does not allow for customization in that model. 
# Monkey patching it is for now.
Embed._meta.get_field("thumbnail_url").max_length = 500
```

After that, run a `makemigrations`, which creates the migration. Then `migrate`. Run this everywhere (staging, prod, dev machines).

When you are upgrading wagtail, you might get a "conflicting migrations" error. Then you delete the migration created from above, upgrade wagtail and run the `makemigrations` process from above again.

It's far from perfect. I hope the team will provide a customization possibility for this soon.

**Comment by allcaps** [comment]
I'll give this a go.

Note that this issue is open for a long time. The devil is in the unique constraint and how that is handled by various databases.
See comment: https://github.com/wagtail/wagtail/issues/1985#issuecomment-469068545 by @ababic :clap:

Option 1: Add a hash field to guarantee uniqueness.
Option 2: Remove unique constraint and validate on clean and save.

**Comment by gasman** [comment]
Fixed in #6635
